src/material-experimental/mdc-slider/slider.ts

```diff
@@ -411,6 +424,22 @@ export class MatSlider implements AfterViewInit, OnChanges, OnDestroy, ControlVa
     }
   }
 
+  getChangeDetectorRef() {
+    return this._changeDetectorRef;
+  }
+
+  getElementRef() {
+    return this._elementRef;
+  }
+
+  getFoundation() {
+    return this._foundation;
+  }
+
+  getNgZone() {
+    return this._ngZone;
+  }
```

Comment:
Could we expose all of these class members by just making them public? We should also prefix them with an underscore as they are used purely internally and should not be part of the public API.

Reply:
If the members are readonly, then that'd be ok. Otherwise, +1 to _names for the methods.

Reply:
Yep, done.

Reply:
Why not just make them readonly public properties when injecting them through the constructor? That would avoid the unnecessary size and source code overhead of these methods?

Reply:
I made _ngZone, _elementRef, and _changeDetectorRef public readonly.